dpcpp/base/executor.dp.cpp

```diff
@@ -150,6 +159,32 @@ int DpcppExecutor::get_num_devices(std::string device_type)
 }
 
 
+bool DpcppExecutor::verify_memory_to(const OmpExecutor *dest_exec) const
+{
+    auto device = detail::get_devices(device_type_)[device_id_];
+
+    return device.is_host() || device.is_cpu();
+}
+
+bool DpcppExecutor::verify_memory_to(const ReferenceExecutor *dest_exec) const
+{
+    auto device = detail::get_devices(device_type_)[device_id_];
+    return device.is_host() || device.is_cpu();
+}
+
+bool DpcppExecutor::verify_memory_to(const DpcppExecutor *dest_exec) const
+{
+    auto device = detail::get_devices(device_type_)[device_id_];
+    auto other_device = detail::get_devices(
+        dest_exec->get_device_type())[dest_exec->get_device_id()];
+    return ((device.is_host() || device.is_cpu()) &&
+            (other_device.is_host() || other_device.is_cpu())) ||
+           (device.get_info<cl::sycl::info::device::device_type>() ==
+                other_device.get_info<cl::sycl::info::device::device_type>() &&
+            device.get() == other_device.get());
```

Comment:
I don't fully understand why you compare both the devices and the device info for equality. Isn't the info always the same when the devices are equal?

Reply:
I am somewhat anticipating machines with multiple Intel GPUs, or even different vendors possibly. In this case, we need to distinguish whether we have an "accelerator", a "gpu" or other, but also if they actually are the same (same ID/vendor combination), or different. I have to say I obviously could not test this piece of code yet, as all machines we have access to only have one GPU.